ovs/lib/helpers/decorators.py

```diff
@@ -122,380 +172,885 @@ def _ensure_single(task_name, mode, extra_task_names=None, global_timeout=300, c
+        :type kwargs_dict: dict
+        :return: The filtered dict
+        :rtype: dict
+        """
+        filtered_kwargs = kwargs_dict.copy()
+        for key in self.ensure_single_container.ignore_arguments:
+            filtered_kwargs.pop(key, None)
+        return filtered_kwargs
+
+    @ensure_run_time_info()
+    @contextmanager
+    def lock_chained_mode(self, kwargs, timeout):
+        """
+        Lock the function in chained mode
+        :param kwargs: Dict containing all arguments as key word arguments
+        :type kwargs: dict
+        :param timeout: Polling timeout in seconds
+        :type timeout: float
+        :raises: EnsureSingleTaskDiscarded: If the task was discarded
+        :raises: EnsureSingleDoCallBack: If the task is required to call the callback function instead of the task function
+        :raises: EnsureSingleSimilarJobsCompleted: If the task was waiting on a similar task to end but those tasks have finished
+        """
+        self.validate_no_extra_names()
+
+        # Update kwargs with args
+        kwargs_dict = self._filter_ignorable_arguments(kwargs)
+        params_info = 'with params {0}'.format(kwargs_dict) if kwargs_dict else 'with default params'
+        task_log_format = 'task {0} {1}'
+        task_log_name = task_log_format.format(self.ensure_single_container.task_name, params_info)
+        task_log_id = task_log_format.format(self.task_id, params_info)
+
+        try:
+            self.run_hook('before_validation')
+            # Validate whether another job with same params is being executed, skip if so
+            # 1st registration is a running job, we check all other queued jobs for identical params
+            self._ensure_job_limit(kwargs_dict, timeout, (task_log_name, task_log_id), job_limit=1, registrations_to_skip=1)
+
+            self.logger.info('New {0} scheduled for execution'.format(task_log_name))
+            new_task_data = {'kwargs': kwargs_dict, 'task_id': self.task_id, 'timestamp': self.now}
+            self._register_task(new_task_data)
+
+            # Poll the arakoon to see whether this call is the first in list, if so --> execute, else wait
+            first_registration = None
+            slept = 0
+            while slept < timeout:
+                current_registrations, initial_registrations = self.get_task_registrations()
+                if current_registrations:
+                    first_registration = current_registrations[0]['timestamp']
+                if first_registration == self.now:
+                    break
+
+                self.unittest_set_state_waiting()
+                slept += self.poll_sleep_time
+                time.sleep(self.poll_sleep_time)
+        finally:
+            self.run_hook('after_validation')
+
+        successful = True
+        try:
+            if slept:
+                self.logger.info('{0} had to wait {1} seconds before being able to start'.format(task_log_name, slept))
+            if slept >= timeout:
```

Comment:
will now log 
` X had to wait Y` before being able to start`
`Could not start Z`
upon timeout. 
```
if slept:
   slept < timeout:
      log dis
   else:
      log dat
```